src/dune_engine/process.ml

```diff
@@ -846,6 +846,14 @@ let report_process_finished
 
 let set_temp_dir_when_running_actions = ref true
 
+let set_temp_dir ~temp_dir env =
+  match temp_dir, !set_temp_dir_when_running_actions with
+  | Some path, _ ->
+    Env.add env ~var:Env.Var.temp_dir ~value:(Path.to_absolute_filename path)
+  | None, true -> Dtemp.add_to_env env
+  | None, false -> env
+;;
```

Comment:
@rgrinberg There's a comment in the code indicating that janestreet uses `set_temp_dir_when_running_actions` internally. Setting it to `false` would accidentally use the correct `$TMPDIR` set by cram in the env, instead of overriding the path with a wrong one. I believe the new code preserves the old behavior, but is there any chance the new `?temp_dir` would help get rid of the ref on your side?

Reply:
I don't see this global variable anymore internally. Feel free to get rid of it here as well I suppose. However, let's try to keep the PR's as focused as possible, so that would work should come in a separate PR.